project.mml

```diff
@@ -444,32 +444,24 @@ Layer:
         (SELECT
             way,
             (CASE WHEN feature IN ('highway_motorway_link', 'highway_trunk_link', 'highway_primary_link', 'highway_secondary_link', 'highway_tertiary_link') THEN substr(feature, 0, length(feature)-4) ELSE feature END) AS feature,
-            horse,
-            foot,
-            bicycle,
             tracktype,
             int_surface,
-            access,
+            int_access,
             construction,
             service,
             link,
             layernotnull
           FROM ( -- subselect that contains both roads and rail
             SELECT
                 way,
-                'highway_' || highway AS feature, --only motorway to tertiary links are accepted later on
-                horse,
-                foot,
-                bicycle,
+                'highway_' || (CASE WHEN highway = 'path' THEN carto_path_type(bicycle, horse) ELSE highway END) AS feature, --only motorway to tertiary links are accepted later on
                 tracktype,
                 CASE WHEN surface IN ('unpaved', 'compacted', 'dirt', 'earth', 'fine_gravel', 'grass', 'grass_paver', 'gravel', 'ground',
                                       'mud', 'pebblestone', 'salt', 'sand', 'woodchips', 'clay', 'ice', 'snow') THEN 'unpaved'
                   WHEN surface IN ('paved', 'asphalt', 'cobblestone', 'cobblestone:flattened', 'sett', 'concrete', 'concrete:lanes',
                                       'concrete:plates', 'paving_stones', 'metal', 'wood', 'unhewn_cobblestone') THEN 'paved'
                 END AS int_surface,
-                CASE WHEN access IN ('destination') THEN 'destination'::text
-                  WHEN access IN ('no', 'private') THEN 'no'::text
-                END AS access,
+                carto_highway_int_access(highway, access, foot, bicycle, horse, tags->'motorcar', tags->'motor_vehicle', tags->'vehicle') AS int_access,
```

Comment:
This reveals an inconsistency of how we're treating access tags. `foot`, `bicycle`, and `horse` get columns while `motorcar`, `motor_vehicle`, and `vehicle` don't.

If we're storing the raw tag text here, let's put them all in the tags column. If we're preprocessing we could have a smaller number of columns

Reply:
Yes, our current database schema has foot, bicycle, and horse as columns (because these have been interpreted in rendering of `highway=path` for a long time) and not the other more specific access tags.  I agree that with this change this distinction would not make that much sense any more and sourcing all of them from hstore would be reasonable.  But since that would involve a database reload i would suggest to do this separately.  Overall, i am not sure if this is worth breaking backwards compatibility of the database.

Reply:
I don't see any issue with doing a reload. Being consistent on columns is a plus, even if we aren't able to reach a consensus on any preprocessing

Reply:
I am fine with removing foot/bicycle/horse from the columns.  I am just not sure if it is worth doing a database layout change breaking backwards compatibility just because of this.

We could also source  foot/bicycle/horse from hstore in anticipation of such a change now and defer the actual removal of the columns to when we do a database reload anyway.

Reply:
> We could also source foot/bicycle/horse from hstore in anticipation of such a change now and defer the actual removal of the columns to when we do a database reload anyway.

This would make sense. But if we are sourcing all the detailed access tags via hstore, I suggest passing the hstore tags as a function argument to both `carto_highway_int_access` and `carto_path_type`. As well as being neater, it is presumably slightly more efficient to selectively access tags as needed rather than unpack them all to pass as function arguments [but I'm no SQL expert]?

Reply:
I have to correct myself here:  I wrongly remembered that tags in columns are also available in hstore, but they are not.  Hence my suggestion would not work.